### PR TITLE
Detect support for _Complex

### PR DESCRIPTION
**Pull request by archi**
Check if the compiler supports _Complex. If not, disable building `libm/complex/` and disable the `complex.h` header.

### CODE REVIEW

**Comment by archi** [comment]
The two fails here seem to be some travis hiccup?

File: newlib/libc/include/complex.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #include <sys/cdefs.h>
 
+#ifdef HAVE_COMPLEX
```
Comment (keith-packard) [inline newlib/libc/include/complex.h:17]: We shouldn't need changes in this file, but we should change the build system to not install it.

**Review by keith-packard — CHANGES_REQUESTED** [review]
Changes look pretty good; let's make it so 'complex.h' doesn't get installed so that applications won't accidentally use it.

**Comment by keith-packard** [comment]
All merged in

**Pull request closed without merging**